### PR TITLE
multiusable keyholes

### PR DESCRIPTION
**Pull request by Nemoel-Tomo**
use a switch trigger for the keyhole to make it multi useable. Lara will get the key back into her inventory after using it. (only with switch triggers)

Using key animation = ocb number

### CODE REVIEW

File: TombEngine/Objects/Generic/puzzles_keys.h [hunk 1]
```diff
@@ -3,6 +3,13 @@
 struct CollisionInfo;
 struct ItemInfo;
 
+enum PuzzleHoleType
```
Comment (Kubsy) [inline TombEngine/Objects/Generic/puzzles_keys.h:6]: should this be KeyHoleType?

**Comment by Stranger1992** [comment]
[multiusable keyholes #1082.zip](https://github.com/MontyTRC89/TombEngine/files/11304073/multiusable.keyholes.1082.zip)

PR exe. Generated 23rd April 2023, 16:16 GTM

File: TombEngine/Objects/Generic/puzzles_keys.h (hunk 1 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/Generic/puzzles_keys.h:6]: this enum is also used by puzzle holes. how should I call it?

Comment (Nemoel-Tomo) [reply TombEngine/Objects/Generic/puzzles_keys.h]: fixed

**Comment by Stranger1992** [comment]
Tested:

- Multi-use puzzle hole ✅
- Multi-use keyhole ✅
- Multi-use keyhole after save/reload ✅
- Multi-use puzzle hole after save/reload ✅
- Multi-use keyhole with missing bits (ie: 1 hole for bit 1-3 and another for 4+5) ✅
-  Multi-use puzzle with missing bits (ie: 1 hole for bit 1-3 and another for 4+5) ✅
- Multi-use keyhole with missing bits. 1 key for 25 puzzle holes. Only triggered when all were used. Use the key again in any keyhole turned off the event being triggered ✅
- Tried OCB 337 (use keycard). Lara still used normal key anim ❌

**Comment by Stranger1992** [comment]
Tried OCB 437 (use keycard). Lara now performs the animation as instructed ✅

**Review by Kubsy — APPROVED** [review]

**Pull request merged**